### ISSUE TITLE
Most efficient way to find oldest metrics?

### ISSUE THREAD

**Issue by HariSekhon** [OpenTSDB/opentsdb#1275]
I've got a chunky OpenTSDB HBase cluster with 330,000 metrics and history going back at least 2.5 years that I can find, so I'm considering deleting history.

What is the most efficient way of finding the age of the oldest metrics to see how old the data is so that I can estimate how much history is reasonable to keep in comparison to the space currently being consumed?

In this case it would be sufficient to just find the very oldest data point across all metrics.

Is there any better way of find this out short of doing a metric by metric scan?

For example I can do something like the following for each metric but this seems expensive and I'd also have to keep tweak the START and END ranges (I already did 4y-ago to 3y-ago and found nothing):
```
tsdb scan --import 3y-ago 2y-ago count df.bytes.total
```
I could probably also do with some refinement to reduce the scan range.

**Comment by HariSekhon** [comment]
FYI I wrote a script in my [PyTools](https://github.com/harisekhon/pytools) repo to get the metric uids along with their creation date. This isn't quite the same thing as it doesn't account for backfill or if metrics newer than the uid creation were deleted, but ok to use on a new cluster and also helped me solve #1294 by finding the metric created at the time when some collector somewhere in another team was sending high cardinality tagv.

**Comment by manolama** [comment]
Yeah there is no really easy way to get the oldest timestamps. You'd have to write some Java code to scan the whole table and print out the first rows for new metrics.
